packages/common/src/content.ts

```diff
@@ -885,6 +891,66 @@ const getSolutionUrl = (content: IHubContent): string => {
   return hubUrl;
 };
 
+/**
+ * Create a new content with updated boundary properties
+ * @param content original content
+ * @param boundary boundary provenance
+ * @returns
+ */
+export const setContentBoundary = (
+  content: IHubContent,
+  boundary: GeographyProvenance
+) => {
+  // update content's item and boundary
+  const properties = { ...(content.item.properties || {}), boundary };
+  const item = { ...content.item, properties };
+  const updated = { ...content, item };
+  return { ...updated, boundary: getContentBoundary(updated) };
+};
+
+/**
+ * Create a new content with updated extent and derived properties like boundary
+ * @param content original content
+ * @param extent new extent
+ * @returns a new content with the updated extent and boundary
+ */
+export const setContentExtent = (
+  content: IHubContent,
+  extent: IBBox
+): IHubContent => {
+  // update content's item and extent
+  const item = { ...content.item, extent };
+  const updated = { ...content, item, extent };
+  // derive boundary from content properties
+  const boundary = getContentBoundary(updated);
+  return { ...updated, boundary };
+};
+
+const getContentBoundary = (content: IHubContent): IHubGeography => {
+  const item = content.item;
+  const extent = item.extent;
+  // user specified provenance is stored in item.properties
+  const provenance: GeographyProvenance =
+    item.properties?.boundary ||
+    // but we default to item if the item has an extent
+    (isExtentCoordinateArray(extent) ? "item" : undefined);
```

Comment:
@drspacemanphd I want to confirm w/ you that this is effectively the same logic for deriving the default `boundary.provenance` that the server uses.

Reply:
I believe so yes. If there is no set boundary in the item properties (either because a content editor has not set it or, potentially in the future, if the server has not already set a default one there), then the default boundary type should be `item`, and the boundary should correspond to the item's extent.

For clarity, I believe the server's get-all-boundaries endpoint returns item extents in GeoJSON format as an envelope with a list of coordinates. I assume that the `itemExtentToBoundary` function is doing something similar.

Reply:
> get-all-boundaries endpoint returns item extents in GeoJSON format as an envelope with a list of coordinates. I assume that the itemExtentToBoundary function is doing something similar.

Good point. It looks like the server returns polygons:

https://hub.arcgis.com/api/v3/datasets/183adc24880b41c4be9fd6a14eb6165f_0

![image](https://user-images.githubusercontent.com/662944/146044968-50b30a2a-7110-4890-b0d9-269ee432053a.png)

but `itemExtentToBoundary()` returns an envelope (xmin, xmax, etc). I should update that to return a polygon.

Reply:
This re-affirms my conviction that the fns from the server should move into Hub.js and the server should consume them from there.